rust_src/src/buffers.rs

```diff
@@ -360,4 +360,25 @@ fn set_buffer(buffer_or_name: LispObject) -> LispObject {
     buffer
 }
 
+/// No such buffer error.
+#[no_mangle]
+pub fn nsberror(spec: LispObject) {
+    if spec.is_string() {
+        let msg = "No buffer named ";
+        let mut argsarray = [
+            unsafe { make_string(msg.as_ptr() as *const c_char, msg.len() as ptrdiff_t) },
+            spec.to_raw(),
+        ];
+        xsignal!(
```

Comment:
Why `xsignal` here when the original called `error` for both cases? You have re-created the `error!` macro.

Reply:
Yeah, I asked in our gitter room a few days ago. And @db48x suggested "well, in that case, error! probably should never call format!". 

You said "@brotzeit what @db48x suggests is also basically what display_string in lisp.rs does. It makes the slice using byte length.".
The problem with `display_string` is that it also makes use of `from_utf8_lossy`, which doesn't work with multi and unibyte chars.

Reply:
I think `error!` isn't the problem. It has to be possible to cast a multi/unibyte string to rust str, but I just don't know how...

Reply:
The C code has not cared. I say port nsberror as written. Then open an issue to add more tests. If we can make the code break then we fix it.

In the C code `nsberror` is just a wrapper around `error`. So it makes sense to fix `error` when and if we find strings that fail there.

Reply:
Yea, don't rewrite error!; it's the right tool for the job. I think we will need to implement our own string type to handle Emacs' internal encoding, but that can be done in a different bug. (See section 22.2 of the Emacs manual and section 32.1 of the Elisp manual.)

Reply:
:+1:

Reply:
So I add a TODO note and we merge or wait until we have a better solution ?

Reply:
add a note and merge

Reply:
Done.